src/services/dex.js

```diff
@@ -1259,112 +1274,118 @@ export default {
       return;
     }
 
+    const currentNetwork = network.getSelectedNetwork();
+
     config.tx.inputs.forEach((input) => {
-      _.set(assetUTXOsToIgnore, input.prevHash, input.prevIndex);
+      _.set(assetUTXOsToIgnore, `${input.prevHash}-${input.prevIndex}-${currentNetwork.net}`, currentNetwork.bestBlock.index);
     });
   },
 
   calculateWithdrawInputsAndOutputs(config, assetId, quantity) {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
```

Comment:
I think it'd be worth it to break this method up into multiple smaller methods. There's A LOT that's going on in here. It's pretty hard to read as it is.

Reply:
By 'this method' I mean, `calculateWithdrawInputsAndOutputs`.

Reply:
Agree. It's tech debt that we will need to prioritize separately.